OpenRA.Game/Network/OrderManager.cs

```diff
@@ -39,6 +39,9 @@ public sealed class OrderManager : IDisposable
 		public string ServerError = null;
 		public bool AuthenticationFailed = false;
 
+		// The default null means "no map restriction" while an empty set means "all maps restricted"
+		public HashSet<string> ServerMapPool = null;
+
```

Comment:
We really need to have a better system to store stuff (with proper lifecycle), now we are storing it here because its convenient.